src/aperture.c

```diff
@@ -430,17 +436,44 @@ int sep_sum_circann_multi(sep_image *im,
                   if (errisstd)
                     varpix *= varpix;
                 }
+              
+              ismasked = 0;
               if (im->mask)
                 {
                   if (mconvert(maskt) > im->maskthresh)
                     {
                       *flag |= SEP_APER_HASMASKED;
                       ismasked = 1;
                     }
-                  else
-                    ismasked = 0;
                 }
 
+              /* Segmentation image:  
+
+    	           If `id` is negative, require segmented pixels within the 
+    	           aperture.
+
+    	           If `id` is positive, mask pixels with nonzero segment ids
+    	           not equal to `id`.
+
+    	      */ 
+    	      if (im->seg)
+      	        {
+      	          if (id > 0) 
+      	            {
+      	              if ((sconvert(segt) > 0.) & (sconvert(segt) != id))
```

Comment:
Since this is in an inner loop, might be helpful for performance to only call `sconvert(segt)` once, and use short-circuiting `&&` instead of bitwise `&`.